Frama-C value plugin reported:

src-linux-3.17.1/arch/um/include/shared/skas/skas.h:11:[value] could not find a size for array
src-linux-3.17.1/include/net/inet_connection_sock.h:186:[value] could not find a size for array

--- a/arch/um/include/shared/skas/skas.h
+++ b/arch/um/include/shared/skas/skas.h
@@ -8,7 +8,7 @@
 
 #include <sysdep/ptrace.h>
 
-extern int userspace_pid[];
+extern int userspace_pid[1];
 extern int proc_mm, ptrace_faultinfo, ptrace_ldt;
 extern int skas_needs_stub;
 
--- a/include/net/inet_connection_sock.h
+++ b/include/net/inet_connection_sock.h
@@ -183,7 +183,7 @@ void inet_csk_delete_keepalive_timer(str
 void inet_csk_reset_keepalive_timer(struct sock *sk, unsigned long timeout);
 
 #ifdef INET_CSK_DEBUG
-extern const char inet_csk_timer_bug_msg[];
+extern const char inet_csk_timer_bug_msg[sizeof("inet_csk BUG: unknown timer value\n")];
 #endif
 
 static inline void inet_csk_clear_xmit_timer(struct sock *sk, const int what)
